### ISSUE TITLE
build: include bundled actors tag somewhere

### ISSUE THREAD

**Issue by arajasek** [filecoin-project/lotus#10185]
We're facing a lot of confusion not knowing what actors bundle release is being run. This information is _technically_ deducable, since the builtin_actors_gen.go file contains the manifest CIDs, which can be cross-compared against the release artifacts in builtin-actors, but that's a lot.

We should modify pack.sh to include this information...somewhere (perhaps builtin_actors_gen.go). Note that this information will not (necessarily) be correct -- no part of the system will be verifying the tag on startup, but that's okay -- this is just for our own sanity.

**Comment by snissn** [comment]
Hey @arajasek - can you provide some extra guidance here? I found the CID associated with the ManifestCid - and I find when I grep my repo it finds that CID in the lotus binary, but beyond that I'm not sure what next steps to take. I also tried to fetch the CID using ipfs but it wasn't downloading.

**Comment by arajasek** [comment]
Yeah, so what I think we should do is incorporate the actual git tag of the bundle being used.

Lotus currently updates / adds bundles with the `pack.sh` script, which takes a git tag as input, and fetches the bundle artifact directly from GitHub. I think it'd be very useful if as part of this process, we _recorded_ the tag somewhere in Lotus itself. This would serve as an easy reference to the Git commit you're running (for example if you wanted to examine the actors source code).

I'm not fussed about _where_ we put this info. The pack.sh script invokes the generator in gen/bundle/bundle.go, which generates builtin_actors_gen.go. It might be possible to include the git tag as input to that generator. Or, if it's easier, just generate a different file with this info.

**Comment by jennijuju** [comment]
so the way we run the ./pack.sh is `./pack.sh <version> <bundle tag>` -  as the script is run, if we can just append the version + bundle into the builtin_actors_gen.go (first line is also okay).

**Comment by snissn** [comment]
Does it make sense to add the git tag to BuiltinActorsMetadata or would the git info be better as a comment in builtin_actors_gen.go?

**Comment by jennijuju** [comment]
> Does it make sense to add the git tag to BuiltinActorsMetadata or would the git info be better as a comment in builtin_actors_gen.go?

Either is fine imho!

**Comment by snissn** [comment]
Great! This PR adds the git tag to BuiltinActorsMetadata  https://github.com/filecoin-project/lotus/pull/10200

I'm trying to test the command but am hitting a snag 
`panic: bundle bundle '"mainnet.tar.zst"' doesn't start with a 'v'`

how should the command be run? i'm guessing something other than `./pack.sh versionmikerstest dev/20230206-pre-rc.6`

**Comment by jennijuju** [comment]
> Great! This PR adds the git tag to BuiltinActorsMetadata #10200
> 
> I'm trying to test the command but am hitting a snag `panic: bundle bundle '"mainnet.tar.zst"' doesn't start with a 'v'`
> 
> how should the command be run? i'm guessing something other than `./pack.sh versionmikerstest dev/20230206-pre-rc.6`

The first param would be the actor version - I.e `v9`, `v10`

**Comment by snissn** [comment]
ok great! The PR looks good to me then, the build appears to succeed when i test it with ./pack.sh v9  dev/20230206-pre-rc.6

**Comment by arajasek** [comment]
Sooooo, we have an issue with 10200 unfortunately -- the `pack.sh` script is a bit more flexible than we budgeted for. In particular, it can be used to update:

- only a specific actors version (the v9 / v10 mentioned by @jennijuju above)
- only a specific _network_ (eg. mainnet / vs. calibrationnet), by providing optional network overrides (eg.  `pack.sh v9 dev/20230114-pre-rc.2 mainnet=v9.0.3` would set all v9 actors to `dev/20230114-pre-rc.2`, EXCEPT mainnet, which would stay at `v9.0.3`).

As currently implemented in 10200, we just set the git tag for all metadata to the single provided input, which is wrong.

We could:

- feed more input into the gen/bundle.go method (perhaps a chain of (<actorsVersion> <network> <tag>) tuples?)
- change the _output_ to not be in `builtin_actors_gen`, but instead somewhere else that we can more easily control directly from `pack.sh`
